docs/Design/Loops/For Each Key.md

```diff
@@ -0,0 +1,182 @@
+# For Each Key
+
+A loop that itterates over a dictionary by key.
+
+## Commands
+
+### `for each key start`
+
+`for each key start : container key string`
+
+Starts a for each key block.
+The first parameter is the container to be itterated over.
+The second parameter is the itterator variable.
+The third parameter is the type of the itterator variable.
```

Comment:
These three sentences are a little bland to read. Can you vary up the sentence structure a bit?